pkg/testing/ess/provisioner.go

```diff
@@ -62,89 +60,77 @@ func (p *provisioner) SetLogger(l runner.Logger) {
+			"deployment_id": resp.ID,
+		},
+		Ready: false,
+	}, nil
+}
 
-	// set a long timeout
-	// this context travels up to the magefile, clients that want a shorter timeout can set
-	// it via mage's -t flag
-	readyCtx, readyCancel := context.WithTimeout(ctx, 25*time.Minute)
-	defer readyCancel()
-
-	g, gCtx := errgroup.WithContext(readyCtx)
-	for req, resp := range results {
-		g.Go(func(req runner.StackRequest, resp *CreateDeploymentResponse) func() error {
-			return func() error {
-				ready, err := p.client.DeploymentIsReady(gCtx, resp.ID, 30*time.Second)
-				if err != nil {
-					return fmt.Errorf("failed to check for cloud %s to be ready: %w", req.Version, err)
-				}
-				if !ready {
-					return fmt.Errorf("cloud %s never became ready: %w", req.Version, err)
-				}
-				return nil
-			}
-		}(req, resp))
+// WaitForReady should block until the stack is ready or the context is cancelled.
+func (p *provisioner) WaitForReady(ctx context.Context, stack runner.Stack) (runner.Stack, error) {
+	deploymentID, err := p.getDeploymentID(stack)
+	if err != nil {
+		return stack, fmt.Errorf("failed to get deployment ID from the stack: %w", err)
 	}
-	err := g.Wait()
+	// allow up to 10 minutes for it to become ready
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+	p.logger.Logf("Waiting for cloud %s to be ready (%s) [id: %s]", stack.Version, stack.ID, deploymentID)
+	ready, err := p.client.DeploymentIsReady(ctx, deploymentID, 30*time.Second)
 	if err != nil {
-		return nil, err
+		return stack, fmt.Errorf("failed to check for cloud %s to be ready: %w", stack.Version, err)
 	}
-
-	var stacks []runner.Stack
-	for req, resp := range results {
-		stacks = append(stacks, runner.Stack{
-			ID:            req.ID,
-			Version:       req.Version,
-			Elasticsearch: resp.ElasticsearchEndpoint,
-			Kibana:        resp.KibanaEndpoint,
-			Username:      resp.Username,
-			Password:      resp.Password,
-			Internal: map[string]interface{}{
-				"deployment_id": resp.ID,
-			},
-		})
+	if !ready {
+		return stack, fmt.Errorf("cloud %s never became ready: %w", stack.Version, err)
 	}
-	return stacks, nil
+	stack.Ready = true
+	return stack, nil
 }
 
-// Clean cleans up all provisioned resources.
-func (p *provisioner) Clean(ctx context.Context, stacks []runner.Stack) error {
-	var errs []error
-	for _, s := range stacks {
-		err := p.destroyDeployment(ctx, s)
-		if err != nil {
-			errs = append(errs, fmt.Errorf("failed to destroy stack %s (%s): %w", s.Version, s.ID, err))
-		}
-	}
-	if len(errs) > 0 {
-		return errors.Join(errs...)
+// Delete deletes a stack.
+func (p *provisioner) Delete(ctx context.Context, stack runner.Stack) error {
+	deploymentID, err := p.getDeploymentID(stack)
+	if err != nil {
+		return err
 	}
-	return nil
+
+	// allow up to 1 minute for request
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	p.logger.Logf("Destroying cloud %s (%s) [id: %s]", stack.Version, stack.ID, deploymentID)
```

Comment:
[SUggestion]
Same here

```suggestion
	p.logger.Logf("Destroying cloud stack %s. StackID: %s, deploymentID: %s", stack.Version, stack.ID, deploymentID)
```

Reply:
Updated.